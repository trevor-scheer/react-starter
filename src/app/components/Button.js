// @flow
import React, {type ComponentType, type Node} from 'react';
import styled from 'styled-components';

type ButtonProps = {
  children: Node,
  onClick?: (SyntheticMouseEvent<HTMLButtonElement>) => void
};

const Button: ComponentType<ButtonProps> = styled.button`
  padding: 8px 16px;
  background-color: #5500bb;
  color: #000000;
`;

export default Button;
