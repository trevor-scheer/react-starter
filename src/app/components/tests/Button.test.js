import React from 'react';
import {shallow} from 'enzyme';
import Button from '../Button';

function renderButton(props = {}) {
  return shallow(<Button {...props} />);
}

describe('Button', () => {
  it('renders', () => {
    expect(renderButton).not.toThrow();
  });

  it('renders with provided text', () => {
    const DUMMY_TEXT = 'Some Text';
    const button = renderButton({children: DUMMY_TEXT});
    expect(button.contains(DUMMY_TEXT)).toBe(true);
    expect(button).toMatchSnapshot();
  });
});
