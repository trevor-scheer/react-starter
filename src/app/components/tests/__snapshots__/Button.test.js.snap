// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with provided text 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.button>
    Some Text
  </styled.button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Some Text",
      "className": "sc-bdVaJa CUsvt",
    },
    "ref": null,
    "rendered": "Some Text",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Some Text",
        "className": "sc-bdVaJa CUsvt",
      },
      "ref": null,
      "rendered": "Some Text",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
